From 222fdde794eee8f78f964a608757ab75a9feb4b0 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Wed, 15 Jan 2014 16:29:09 -0800
Subject: [PATCH 062/456] drm: xilinx: drv: Add xilinx_drm_format_bpp()

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Add xilinx_drm_format_bpp() which returns bpp of the given format.

Signed-off-by: Hyun Kwon <hyunk@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 63e1ee543792296caf53a8ec66924b5ded0691f9)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/gpu/drm/xilinx/xilinx_drm_drv.c |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
index 5dd5566..664c0ed 100644
--- a/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
+++ b/drivers/gpu/drm/xilinx/xilinx_drm_drv.c
@@ -175,6 +175,23 @@ int xilinx_drm_format_by_name(const char *name, uint32_t *drm_format)
 	return -EINVAL;
 }
 
+/* get bpp of given format */
+static unsigned int xilinx_drm_format_bpp(uint32_t drm_format)
+{
+	const struct xilinx_video_format_desc *format;
+	unsigned int i;
+
+	for (i = 0; i < ARRAY_SIZE(xilinx_video_formats); i++) {
+		format = &xilinx_video_formats[i];
+		if (format->drm_format == drm_format)
+			return format->bpp;
+	}
+
+	DRM_ERROR("Unknown/unsupported DRM format: %d\n", drm_format);
+
+	return -EINVAL;
+}
+
 /* load xilinx drm */
 static int xilinx_drm_load(struct drm_device *drm, unsigned long flags)
 {
-- 
1.7.5.4

